main.go

```diff
@@ -277,8 +283,72 @@ func (p *prometheus) Collect(ch chan<- registry.Metric) {
 	}
 }
 
+func usage() {
+	groups := make(map[string][]*flag.Flag)
+	// Set a default group for ungrouped flags.
+	groups["."] = make([]*flag.Flag, 0)
+
+	// Bucket flags into groups based on the first of their dot-separated levels.
+	flag.VisitAll(func(fl *flag.Flag) {
+		parts := strings.SplitN(fl.Name, ".", 2)
+		if len(parts) == 1 {
+			groups["."] = append(groups["."], fl)
+		} else {
+			name := parts[0]
+			groups[name] = append(groups[name], fl)
+		}
+	})
+
+	groupsOrdered := make(sort.StringSlice, 0, len(groups))
+	for groupName := range groups {
+		groupsOrdered = append(groupsOrdered, groupName)
+	}
+	sort.Sort(groupsOrdered)
+
+	fmt.Fprintf(os.Stderr, "Usage: %s [options ...] -config.file=<config_file>:\n\n", os.Args[0])
```

Comment:
I don't consider it special treatment but rather additional emphasis.
There are many flags by now, most of which are for tweaking behavior but the config file is the most crucial one.

I think this achieves the goal of showing new users how to start Prometheus without crashing immediately, i.e. "provide a config file through -config.flag".
The list of 30+ flags below is something you only want to scan through if you know what you are looking for.